daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Transaction.scala

```diff
@@ -103,8 +102,8 @@ object VersionedTransaction extends value.CidContainer2[VersionedTransaction] {
   * For performance reasons, users are not required to call `isWellFormed`.
   * Therefore, it is '''forbidden''' to create ill-formed instances, i.e., instances with `!isWellFormed.isEmpty`.
   */
-final private[lf] case class GenTransaction[Nid, +Cid, +Val](
-    nodes: HashMap[Nid, GenNode[Nid, Cid, Val]],
+case class GenTransaction[Nid, +Cid, +Val](
```

Comment:
```suggestion
final case class GenTransaction[Nid, +Cid, +Val](
```
Any reason to not leave it `final`?